### ISSUE TITLE
Dask breaks the TF gradient tape for parallel=True

### ISSUE THREAD

**Issue by antalszava** [PennyLaneAI/pennylane#504]
#### Issue description

One of the tests for `dask` in the `test_qnode_collection.py` behaves differently than expected.

* *Expected behavior:*
The [tests for grad_tf](https://github.com/XanaduAI/pennylane/blob/719e27c7d046e7d488b8c68685fe803f4f5afa9f/tests/collections/test_qnode_collection.py#L243) are currently expected to raise an `AttributeError` both for `parallel=False` and `parallel=True`.

* *Actual behavior:*
The test for `parallel=False` passes.

* *Reproduces how often:* (What percentage of the time does it reproduce?)
For multiple runs of the test suite.

* *System information:*
```
WARNING: pip is being invoked by an old script wrapper. This will fail in a future version of pip.
Please see https://github.com/pypa/pip/issues/5599 for advice on fixing the underlying issue.
To avoid this problem you can invoke Python with '-m pip' instead of running pip directly.
Name: PennyLane
Version: 0.9.0.dev0
Summary: PennyLane is a Python quantum machine learning library by Xanadu Inc.
Home-page: https://github.com/XanaduAI/pennylane
Author: None
Author-email: None
License: Apache License 2.0
Location: /home/antal/xanadu/pennylane
Requires: numpy, scipy, networkx, autograd, toml, appdirs, semantic-version
Required-by: PennyLane-SF, PennyLane-Forest, PennyLane-Qchem, PennyLane-PQ, PennyLane-qsharp, PennyLane-Cirq
Platform info:           Linux-4.19.11-041911-generic-x86_64-with-debian-buster-sid
Python version:          3.7.6
Numpy version:           1.16.4
Scipy version:           1.4.1
Installed devices:
- default.gaussian (PennyLane-0.9.0.dev0)
- default.qubit (PennyLane-0.9.0.dev0)
- default.tensor (PennyLane-0.9.0.dev0)
- default.tensor.tf (PennyLane-0.9.0.dev0)
- strawberryfields.fock (PennyLane-SF-0.6.0)
- strawberryfields.gaussian (PennyLane-SF-0.6.0)
- forest.numpy_wavefunction (PennyLane-Forest-0.6.0)
- forest.qpu (PennyLane-Forest-0.6.0)
- forest.qvm (PennyLane-Forest-0.6.0)
- forest.wavefunction (PennyLane-Forest-0.6.0)
- projectq.classical (PennyLane-PQ-0.7.0.dev0)
- projectq.ibm (PennyLane-PQ-0.7.0.dev0)
- projectq.simulator (PennyLane-PQ-0.7.0.dev0)
- microsoft.QuantumSimulator (PennyLane-qsharp-0.6.0)
- cirq.simulator (PennyLane-Cirq-0.8.0)
None
```

**Comment by josh146** [comment]
Note: the pytorch gradient tape appears to work properly when using Dask; this issue is specific to TensorFlow. Specifically, using Dask in threaded mode seems to cause unconnected gradient errors.

**Comment by antalszava** [comment]
The action item here would be to investigate why `parallel=False` doesn't fail. This would also likely move us closer to understanding if there was a way around breaking the TF gradient tape when `parallel=True`.

QNodeCollections have been considered for refactoring and this issue has not received attention since its opening. Therefore it seems fit to add the `won't fix` label here.

**Comment by josh146** [comment]
@antalszava I wonder if this now works, as since the time this issue was opened, we completely have replaced the internal interface logic 🤔

**Comment by antalszava** [comment]
I believe @dime10 gave this a go and mentioned that the behaviour still exists.

**Comment by josh146** [comment]
My dream of closing issues because they have magically become fixed by the passage of time remains unfulfilled 😢

> QNodeCollections have been considered for refactoring and this issue has not received attention since its opening. Therefore it seems fit to add the won't fix label here.

Yes, I would add `won't fix` here, since we are likely going to remove this built-in Dask integration, and instead show examples of how to set up Dask for parallelization explicitly.

**Comment by trbromley** [comment]
> My dream of closing issues because they have magically become fixed by the passage of time remains unfulfilled 😢

Can your dream be fulfilled now? We no longer have `QNodeCollection`s :tada:
